### DISCUSSION TITLE
Looking for help with writing test cases.

### DISCUSSION THREAD

**Discussion by DungeonMaestro215** [hneemann/Digital#928] [Q&A] [answered]
Howdy. 

I am trying to test a RAM module by storing randomly generated values into it and then reading them back out. There are 16 addresses, each of which can store 4 bits. I want to generate the values, store them all into the RAM, and then check them all; however, I do not know the best way of doing this. I have looked at the included documentation, but I was wondering if this is based on a language I can look for on the web. 

Is there a way to make arrays? Also, is there a way to get the last 4 bits of a value to check it against an output? (I am aware of the 'bits' function, but I believe that this splits it up so that I could check 4 1-bit outputs, not 1 4-bit output.) 

Thanks for any help, and I would be happy to include an example file if that would be helpful.

**Comment by hneemann** [comment, marked as answer]
Arrays cannot be used. So you have to use computable values that depend on an index. But this is not significantly worse than random values.
